File: Insert Date.lua
--[[
VPLanguage = lua
VPScriptMenuTitle = Insert Date
VPShortcutKey = J
VPShortcutMask = control
VPEndConfig
--]]

-- check out http://www.lua.org/pil/22.1.html for more formatting options.

textView = windowController:textView()
document = windowController:document()

dateFormat = os.date("%Y%m%d", os.time())

if textView ~= nil then
    textView:insertText(dateFormat)
end

File: Insert Date.py
# -*- coding: utf-8 -*-

'''
:Title
Insert Date

:Planguage
Python

:Requires
VoodooPad 3.5+

:Description
Inserts Date

EOD
'''
VPScriptSuperMenuTitle = "GTD"
VPScriptMenuTitle = "Insert Date"
VPShortcutMask = "control"
VPShortcutKey = "J"

import AppKit
import time

def main(windowController, *args, **kwargs):
    textView = windowController.textView()
    document = windowController.document()

    if textView != None:
        dateFormat = time.strftime("%Y%m%d")
        textView.insertText_(dateFormat)

